lib/dor/text_extraction/abbyy/results.rb

```diff
@@ -47,6 +55,25 @@ def alto_doc
         def xml_contents
           @xml_contents ||= Nokogiri::XML(File.read(result_path))
         end
+
+        # Software and version used to do OCR processing, found in ALTO output
+        # rubocop:disable Metrics/AbcSize
+        def processing_metadata
+          alto_xml = Nokogiri::XML(File.read(alto_doc)).remove_namespaces!
+          metadata = alto_xml.xpath('//OCRProcessing//ocrProcessingStep').first
+          software_name = metadata.xpath('//softwareName').text
+          software_version = metadata.xpath('//softwareVersion').text
+          return {} if software_name.empty? || software_version.empty?
+
+          # Set class variables so other instances can use them
+          self.class.last_software_name = software_name
+          self.class.last_software_version = software_version
+
+          { software_name:, software_version: }
+        rescue StandardError
+          {}
```

Comment:
this will happen for every PDF, because it doesn't have an ALTO xml from which to get this metadata: line 1 will error because we can't read the (nonexistent) XML file.

admittedly not great practice to do the blanket `rescue`, but it seemed reasonable to be like "if parsing goes wrong for any reason, just use the memorized value" without needing to enumerate the possibilities (file can't load, XML parsing fails, missing XML element...etc)

Reply:
I guess maybe it's two categories:
- missing ALTO XML? that's expected, just return `{}` and move on
- anything else? not expected; log to honeybadger and return `{}`

So, maybe I just need a separate `rescue` for `FileNotFoundError` or something.

Reply:
> this will happen for every PDF, because it doesn't have an ALTO xml from which to get this metadata: line 1 will error because we can't read the (nonexistent) XML file.

ah, ok, that's less exceptional that i was imagining, so i think my suggestion would be uselessly noisy.
 
> I guess maybe it's two categories:
> - missing ALTO XML? that's expected, just return `{}` and move on
> - anything else? not expected; log to honeybadger and return `{}`
> 
> So, maybe I just need a separate `rescue` for `FileNotFoundError` or something.

i like that idea.  could also just leave a comment explaining why the blanket `rescue` is reasonable.  no strong feelings from me.